static/src/javascripts/projects/common/modules/identity/upsell/store/consents.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+import {
+    getAllConsents,
+    getUserFromApi,
+    setConsent,
+} from 'common/modules/identity/api';
+
+type Consent = {
+    id: string,
+    name: string,
+    description: string,
+    isOptOut: boolean,
+    isChannel: boolean,
+    hasConsented: ?boolean,
```

Comment:
I think we should have two types, `ConsentType`, and `Consent`, which is just a pair of `hasConsented` and a `ConsentType`: this will allow code to operate on the two independently... what do you think?

Reply:
Also not sure how closely this aligns with the 'inversion of state' stuff you showed me earlier ;)

Reply:
it needs to be optional because consents might be null for real :D (even if we don't honor this right now)